backends/events/sdl/sdl-events.cpp

```diff
@@ -490,28 +480,57 @@ Common::KeyCode SdlEventSource::SDLToOSystemKeycode(const SDLKey key) {
 	case SDLK_F13: return Common::KEYCODE_F13;
 	case SDLK_F14: return Common::KEYCODE_F14;
 	case SDLK_F15: return Common::KEYCODE_F15;
-	case SDLK_NUMLOCK: return Common::KEYCODE_NUMLOCK;
 	case SDLK_CAPSLOCK: return Common::KEYCODE_CAPSLOCK;
-	case SDLK_SCROLLOCK: return Common::KEYCODE_SCROLLOCK;
 	case SDLK_RSHIFT: return Common::KEYCODE_RSHIFT;
 	case SDLK_LSHIFT: return Common::KEYCODE_LSHIFT;
 	case SDLK_RCTRL: return Common::KEYCODE_RCTRL;
 	case SDLK_LCTRL: return Common::KEYCODE_LCTRL;
 	case SDLK_RALT: return Common::KEYCODE_RALT;
 	case SDLK_LALT: return Common::KEYCODE_LALT;
-	case SDLK_LSUPER: return Common::KEYCODE_LSUPER;
-	case SDLK_RSUPER: return Common::KEYCODE_RSUPER;
 	case SDLK_MODE: return Common::KEYCODE_MODE;
-	case SDLK_COMPOSE: return Common::KEYCODE_COMPOSE;
 	case SDLK_HELP: return Common::KEYCODE_HELP;
-	case SDLK_PRINT: return Common::KEYCODE_PRINT;
 	case SDLK_SYSREQ: return Common::KEYCODE_SYSREQ;
-#if !SDL_VERSION_ATLEAST(2, 0, 0)
-	case SDLK_BREAK: return Common::KEYCODE_BREAK;
-#endif
 	case SDLK_MENU: return Common::KEYCODE_MENU;
 	case SDLK_POWER: return Common::KEYCODE_POWER;
 	case SDLK_UNDO: return Common::KEYCODE_UNDO;
+#if SDL_VERSION_ATLEAST(2, 0, 0)
+	case SDLK_KP_0: return Common::KEYCODE_KP0;
+	case SDLK_KP_1: return Common::KEYCODE_KP1;
+	case SDLK_KP_2: return Common::KEYCODE_KP2;
+	case SDLK_KP_3: return Common::KEYCODE_KP3;
+	case SDLK_KP_4: return Common::KEYCODE_KP4;
+	case SDLK_KP_5: return Common::KEYCODE_KP5;
+	case SDLK_KP_6: return Common::KEYCODE_KP6;
+	case SDLK_KP_7: return Common::KEYCODE_KP7;
+	case SDLK_KP_8: return Common::KEYCODE_KP8;
+	case SDLK_KP_9: return Common::KEYCODE_KP9;
+	case SDLK_NUMLOCKCLEAR: return Common::KEYCODE_NUMLOCK;
+	case SDLK_SCROLLLOCK: return Common::KEYCODE_SCROLLOCK;
+	case SDLK_LGUI: return Common::KEYCODE_LSUPER;
+	case SDLK_RGUI: return Common::KEYCODE_RSUPER;
+	case SDLK_APPLICATION: return Common::KEYCODE_COMPOSE;
+	case SDLK_PRINTSCREEN: return Common::KEYCODE_PRINT;
+	case SDL_SCANCODE_TO_KEYCODE(SDL_SCANCODE_GRAVE): return Common::KEYCODE_TILDE;
+#else
+	case SDLK_KP0: return Common::KEYCODE_KP0;
+	case SDLK_KP1: return Common::KEYCODE_KP1;
+	case SDLK_KP2: return Common::KEYCODE_KP2;
+	case SDLK_KP3: return Common::KEYCODE_KP3;
+	case SDLK_KP4: return Common::KEYCODE_KP4;
+	case SDLK_KP5: return Common::KEYCODE_KP5;
+	case SDLK_KP6: return Common::KEYCODE_KP6;
+	case SDLK_KP7: return Common::KEYCODE_KP7;
+	case SDLK_KP8: return Common::KEYCODE_KP8;
+	case SDLK_KP9: return Common::KEYCODE_KP9;
+	case SDLK_NUMLOCK: return Common::KEYCODE_NUMLOCK;
+	case SDLK_SCROLLOCK: return Common::KEYCODE_SCROLLOCK;
+	case SDLK_LSUPER: return Common::KEYCODE_LSUPER;
+	case SDLK_RSUPER: return Common::KEYCODE_RSUPER;
+	case SDLK_COMPOSE: return Common::KEYCODE_COMPOSE;
+	case SDLK_PRINT: return Common::KEYCODE_PRINT;
+	case SDLK_BREAK: return Common::KEYCODE_BREAK;
+	case SDLK_WORLD_16: return Common::KEYCODE_TILDE;
+#endif
```

Comment:
To be perfectly honest, I prefer the keycode typedefs for SDL1.2 <-> SDL2 than this. It looks uglier, and the code is duplicated. Could we please readd the keycode typedefs? :)